iis-workflows/iis-workflows-ingest-pmc/src/main/java/eu/dnetlib/iis/workflows/ingest/pmc/metadata/MetadataImporter.java

```diff
@@ -46,15 +85,29 @@ protected void map(AvroKey<DocumentText> key, NullWritable value, Context contex
 				saxParser.parse(new InputSource(
 						new StringReader(nlm.getText().toString())), 
 						pmcXmlHandler);
-                context.write(new AvroKey<ExtractedDocumentMetadata>(output.build()), 
-                		NullWritable.get());
+				mos.write(namedOutputMeta, 
+						new AvroKey<ExtractedDocumentMetadata>(output.build()));
         	 } catch (ParserConfigurationException e) {
-             	throw new IOException("Text extraction failed for id " + nlm.getId() + 
-             			" and text: " + nlm.getText(), e);
+        		 Map<CharSequence, CharSequence> auditSupplementaryData = new HashMap<CharSequence, CharSequence>();
+        			auditSupplementaryData.put(FAULT_TEXT, nlm.getText());
```

Comment:
I am also fine with this solution as soon as everyone remembers about taking care of updating this comment whenever set of supported keys changes. How can we guarantee this? 

Besides, currently only theoretically, If we want to keep posted all the people depending on  `iis-schema` module lib in their dependencies this will also require us to release this module only due to new comment was added.